Information model section update (#44)
IETF-OPSAWG-WG/draft-opsarea-rfc5706bis: draft-opsarea-rfc5706bis.md

   entity.

   Information models are helpful to try to focus interoperability on
   the semantic level -- they define what information
   should be gathered and how that information might be used,
   regardless of which management interface carries the data or which

>    vendor product produces the data. The use of an information model might

Reviewer: ```suggestion
   vendor implementation produces the data. The use of an information model might
```

   help improve the ability of operators to correlate messages in
   different protocols where the data overlaps, such as a YANG data model
   and IPFIX Information Elements about the same event. An information model
   might identify which error conditions should be counted separately,
   and which error conditions can be recorded together in a single

>    counter. Then, whether the counter is gathered via Netconf/YANG or

Reviewer: ```suggestion
   counter. Then, whether the counter is gathered via, e.g., NETCONF/YANG or
```

>    counter. Then, whether the counter is gathered via Netconf/YANG or 

Reviewer 2: NETCONF

Author: ACK

Author: Done

   DM        DM         DM     --> concrete/detailed model
                                      for implementers

~~~~
{: #fig-im-dm title="Information Models（IMs） and Data Models（DMs）" artwork-align="center"}

>    Protocol designers may decide an information model to describe the managed elements in a new protocol or protocol extension. The protocol designers may then use the information model to develop data models that will be used for managing the protocols. Alternatively, the protocol designers may directly develop data models without first producing an information model.

Reviewer: ```suggestion
   Protocol designers may decide to use an information model to describe the managed elements in a protocol or protocol extension. The protocol designers may then use the information model to develop data models that will be used for managing the protocols. Alternatively, the protocol designers may directly develop data models without first producing an information model. For example, such a decision can be taken when it is given that the data component is not used by distinct protocols (e.g., IPFIX-only).
```

>    Protocol designers may decide an information model to describe the managed elements in a new protocol or protocol extension. The protocol designers may then use the information model to develop data models that will be used for managing the protocols. Alternatively, the protocol designers may directly develop data models without first producing an information model. 

Reviewer 2: Maye decide to _to include_?  Something is missing from this first sentence.  Perhaps even, "Protocol designers may use an information model to describe the managed elements in a new protocol or protocol extension."

Reviewer 2: And maybe "managing the protocol" with protocol singular is a better choice here.

Author: Yes, I posted it wrong on GitHub. The original sentence is: may decide to use an information model... Is this appropriate?

Reviewer 2: That works.

Author: Done

>    "On the Difference between Information Models and Data Models"

Reviewer 3: ```suggestion
```

>    {{?RFC3444}} is helpful in determining what information to consider

Reviewer 3: ```suggestion
```

>    regarding information models (IMs), as compared to data models (DMs).

Reviewer 3: ```suggestion
```

> Protocol designers should develop an information model if multiple data model representations (e.g. YANG [RFC7950], IPFIX [RFC7011], gMNI [gMNI], etc.) are to be produced, to ensure lossless semantic mapping. Protocol designers may also make an information model if the consequent data models will be complex or many.

Reviewer: ```suggestion
Protocol designers should develop an information model if multiple data model representations (e.g., YANG {{?RFC6020}}{{?RFC7950}} and IPFIX {{?RFC7011}}) are to be produced, to ensure lossless semantic mapping. Protocol designers may also use an information model if the data models will be complex.
```

can we have an example of the last part?

Author: The intent here is to suggest that an information model can be helpful when the data model becomes complex — for example, when it spans multiple related YANG modules. A unified and simplified information model can assist protocol and data model designers by providing a clear architectural view. Now, finding an example may be hard. Perhaps RFC 8328 which references the (expired) draft-ietf-supa-generic-policy-info-model
Alternatively, RFC 8637 is referenced by draft-ietf-teas-yang-path-computation and draft-yu-ccamp-resource-pm-yang

Author: Joe suggest, quoted above:： "Protocol designers may also create an information model if the resulting data models are complex or numerous."

> Protocol designers should develop an information model if multiple data model representations (e.g. YANG [RFC7950], IPFIX [RFC7011], gMNI [gMNI], etc.) are to be produced, to ensure lossless semantic mapping. Protocol designers may also make an information model if the consequent data models will be complex or many. 

Reviewer 2: These bracket callouts should be {{?}} references.

Reviewer 2: And perhaps, "Protocol designers may also create an information model if the resulting data models are complex or numerous."

Author: These bracket callouts should be {{?}} references.
---Yes.

And perhaps, "Protocol designers may also create an information model if the resulting data models are complex or numerous."
---I agree.

Author: Done

> Protocol designers should develop an information model if multiple data model representations (e.g., YANG {{?RFC6020}}{{?RFC7950}} and IPFIX {{?RFC7011}}) are to be produced, to ensure lossless semantic mapping. Protocol designers may create an information model if the resulting data models are complex or numerous.

Reviewer 4: ```suggestion
Protocol designers should develop an information model if multiple data model representations (e.g., YANG {{?RFC6020}}{{?RFC7950}} and/or IPFIX {{?RFC7011}}) are to be produced, to ensure lossless semantic mapping. Protocol designers may create an information model if the resulting data models are complex or numerous.
```

> Protocol designers should develop an information model if multiple data model representations (e.g., YANG {{?RFC6020}}{{?RFC7950}} and/or IPFIX {{?RFC7011}}) are to be produced, to ensure lossless semantic mapping. Protocol designers may create an information model if the resulting data models are complex or numerous.

Reviewer 3: ```suggestion
```

   This document recommends keeping the information model as simple as
   possible by applying the following criteria:

   1.  Start with a small set of essential objects and make additions only as

>        further objects are needed with the objective of keeping the total number of objects as small as possible.

Reviewer: ```suggestion
       further objects are needed with the objective of keeping the total number of objects as small as possible.
```

How "keeping the total number of objects as small as possible" can be characterized?

Author: This is meant as an absolute design objective: to minimize the number of objects while still meeting all functional requirements. 
We can consider rewording the sentence to something like:
“keeping the absolute number of objects as small as possible while still delivering the required function such that there is no duplication between objects and where one piece of information can be derived from the other pieces of information, it is not itself represented as an object.”

Author: Text updated

   2.  Require that all objects be essential for management.

>    3.  Consider evidence of current use of the managed protocol, and the perceived utility of objects added to the information model.

Reviewer: We may clarify why "of the managed protocol" is mentioned here.

Author: The point here is that the choice of what objects to add should be influenced by experience with the protocol (where such experience is available) because the experience should help us know what needs to be managed and what information can be usefully gathered.

   4.  Exclude objects that can be derived from others in this or
       other information models.

   5.  Avoid causing critical sections to be heavily instrumented. A
       guideline is one counter per critical section per layer.

>    6.  When defining an information model in YANG [RFC7950] with Data Structure Extensions [RFC8791] (thereby keeping it abstract and implementation-agnostic per [RFC3444]) ensure that the module remains simple, modular, and clear by following the authoring guidelines in [I-D.ietf-netmod-rfc8407bis].

Reviewer: ```suggestion
   6.  When defining an information model using  YANG Data Structure Extensions {{?RFC8791}} (thereby keeping it abstract and implementation-agnostic per {{?RFC3444}}) ensure that the module remains simple, modular, and clear by following the authoring guidelines in {{?I-D.ietf-netmod-rfc8407bis}}.
```

>   7.  When illustrating the abstract information model, use the YANG tree diagram from RFC 8340 to provide a simple, standardized, and implementation-neutral model structure.

Reviewer: ```suggestion
  7.  When illustrating the abstract information model, use YANG Tree Diagrams {{?RFC8340}} to provide a simple, standardized, and implementation-neutral model structure.
```

   This document recommends keeping the information model as simple as
   possible by applying the following criteria:

   1.  Start with a small set of essential objects and make additions only as
       further objects are needed with the objective of keeping the total number of objects as small as possible. .

   2.  Require that all objects be essential for management.

   3.  Consider evidence of current use of the managed protocol, and the perceived utility of objects added to the information model.

   

   4.  Exclude objects that can be derived from others in this or
       other information models.

   5.  Avoid causing critical sections to be heavily instrumented. A
       guideline is one counter per critical section per layer.

>    6.  When defining an information model in YANG [RFC7950] with Data Structure Extensions [RFC8791] (thereby keeping it abstract and implementation-agnostic per [RFC3444]) ensure that the module remains simple, modular, and clear by following the authoring guidelines in [I-D.ietf-netmod-rfc8407bis].

Reviewer 2: More references rather than raw bracket notation.  And instead of "module" I think "information model" .

Author: Thanks and updated as follows：When defining an information model in YANG {{?RFC7950}} with Data Structure Extensions {{?RFC8791}} (thereby keeping it abstract and implementation-agnostic per {{?RFC3444}}), ensure that the information model remains simple, modular, and clear by following the authoring guidelines in {{?I-D.ietf-netmod-rfc8407bis}}.

Reviewer 2: That works for me.

Author: Done

>   7.  When illustrating the abstract information model, use the YANG tree diagram from RFC 8340 to provide a simple, standardized, and implementation-neutral model structure. 

Reviewer 2: Reference syntax needed.

Author: Thanks and updated as follows： {{?RFC8340}}

Author: Done
